scripts/verify-jenkins.sh

```diff
@@ -0,0 +1,230 @@
+	fi
+}
+
+# verify_plugins()
+#
+# verifies that the plugins from the bundle-plugins.txt file exist and that they
+# are the at version specified by the bundle-plugins.txt file in the code repository 
+# at the specified SHA
+#
+verify_plugins() {
+	while IFS= read -r plugin
+	do
+		if [[ "$plugin" =~ "Generated" ]]; then
+			continue
+		fi
+		name="${plugin%%:*}"
+		version="${plugin##*:}"
+
+		printf "Verifying %s/%s.jpi\n" "${PLUGINS_DIR}" "${name}"
+		printf "     - checking if ${PLUGINS_DIR}/%s.jpi exists ... " "$name"
+		if test -f "${PLUGINS_DIR}/${name}.jpi"; then
+			printf "yes\n"
+		else
+			printf "no\n"
+			echo "${PLUGINS_DIR}/${name}.jpi does not exist but should" >> $FAILED_LOG_LOCATION
+		fi
+
+		found_version=$(unzip -p "${PLUGINS_DIR}/${name}.jpi" META-INF/MANIFEST.MF | grep "^Plugin-Version: " | sed -e 's#^Plugin-Version: ##')
+		found_version=${version%%[[:space:]]}
+		printf "     - checking if version matches %s ... " "$version"
+		if [ "$version" == "$found_version" ]; then
+			printf "yes\n"
+		else
+			printf "no\n"
+			echo " ${PLUGINS_DIR}/${plugin} should be ${version} but was ${found_version} instead" >> $FAILED_LOG_LOCATION
+		fi
+	done < "$COMMIT_BUNDLE_PLUGINS_LOCATION"
+}
+
+# verify_jenkins_war()
+#
+# verifies that the /usr/lib/jenkins/jenkins.war file exists and is at the
+# same version specified by the jenkins-version.txt file at the provided SHA
+#
+verify_jenkins_war() {
+	printf "Verifying %s:\n" "$JENKINS_WAR_LOCATION"
+	jenkins_version=$(cat "$COMMIT_JENKINS_VERSION_LOCATION")
+	jenkins_version_found=$(/usr/lib/jvm/java-11/bin/java -jar "$JENKINS_WAR_LOCATION" --version 2> /dev/null)
+	printf "     - checking if %s exists ... " "$JENKINS_WAR_LOCATION"
+		if test -f "$JENKINS_WAR_LOCATION"; then
+			printf "yes\n"
+		else
+			printf "no\n"
+			printf "%s does not exist" "$JENKINS_WAR_LOCATION" >> $FAILED_LOG_LOCATION
+		fi
+	printf "     - checking if version matches %s ... " "$jenkins_version"
+	if [[ "$jenkins_version_found" == "$jenkins_version" ]]; then
+		printf "yes\n"
+	else
+		printf "no\n"
+		echo "wanted Jenkins version ${jenkins_version} but found ${jenkins_version_found} instead" >> $FAILED_LOG_LOCATION
+	fi
+}
+
+# verify_jenkins_startup()
+#
+# verifies that Jenkins starts up without any plugin conflicts
+# Jenkins will ultimately fail to start correctly but enough of the process
+# completes that we can check for this specific set of issues
+#
+verify_jenkins_startup() {
+	if test -f $JENKINS_START_SCRIPT_LOCATION; then
+		printf "Starting Jenkins with /usr/libexec/s2i/bin\n"
+		printf "This could take up to 60 seconds\n"
+		$JENKINS_START_SCRIPT_LOCATION &> /tmp/verify-jenkins.log &
+		for i in {1..60}
+		do
+			printf "."
+			sleep 1
+		done
+		printf "Parsing log file for plugin conflicts ...\n"
+		for e in "${ERROR_STRINGS[@]}"; do
+			printf "searching log file for error '%s'\n" "$e"
+			FOUND=$(grep jenkins.log -e "$e" -c)
```

Comment:
This greps the ERROR_STRINGS from a file named `jenkins.log`, whereas we are redirecting the stdout & stderr to `/tmp/verify-jenkins.log`. Should the file name be `/tmp/verify-jenkins.log` while performing grep? Because it reported  the following error:

> grep: jenkins.log: No such file or directory
> searching log file for error 'Failed Loading plugin'
> grep: jenkins.log: No such file or directory
>

Reply:
Excellent catch, I missed that when I was doing a refactor. Thank you!